### PR TITLE
Fixed for jira issue - APPM-1058 

### PR DESCRIPTION
**Pull request by lakshani**

### CODE REVIEW

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java [hunk 1]
```diff
@@ -115,12 +115,21 @@ public boolean createProvider(SSOProvider provider) {
...
     public boolean createProvider(WebApp webApp) {
+        String acsUrl = webApp.getAcsURL().trim();
         SSOProvider ssoProvider = webApp.getSsoProviderDetails();
         if (ssoProvider == null) {
             log.warn("No SSO Configurator details given. Manual setup of SSO Provider required.");
         }
 
-        ssoProvider.setAssertionConsumerURL(SSOConfiguratorUtil.getGatewayUrl(webApp));
+        if(acsUrl != null) {
+            if (acsUrl.length() > 0) {
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java]: Put comment

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java [hunk 2]
```diff
@@ -115,12 +115,21 @@ public boolean createProvider(SSOProvider provider) {
...
     @Override
     public boolean createProvider(WebApp webApp) {
+        String acsUrl = webApp.getAcsURL().trim();
         SSOProvider ssoProvider = webApp.getSsoProviderDetails();
         if (ssoProvider == null) {
             log.warn("No SSO Configurator details given. Manual setup of SSO Provider required.");
         }
 
-        ssoProvider.setAssertionConsumerURL(SSOConfiguratorUtil.getGatewayUrl(webApp));
+        if(acsUrl != null) {
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java:124]: StringUtils.isEmpty

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java [hunk 3]
```diff
@@ -188,13 +197,22 @@ public boolean updateProvider(SSOProvider provider) {
...
     public boolean updateProvider(WebApp application) {
+        String acsUrl = application.getAcsURL().trim();
         SSOProvider ssoProvider = application.getSsoProviderDetails();
         if (ssoProvider == null) {
             log.warn("No SSO Configurator details given. Manual setup of SSO Provider required.");
         }
 
         //ssoProvider.setIssuerName(app.getId().getApiName());
-        ssoProvider.setAssertionConsumerURL(SSOConfiguratorUtil.getGatewayUrl(application));
+        if(acsUrl != null) {
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java:207]: Use StringUtil

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java [hunk 4]
```diff
@@ -188,13 +197,22 @@ public boolean updateProvider(SSOProvider provider) {
 
     @Override
     public boolean updateProvider(WebApp application) {
+        String acsUrl = application.getAcsURL().trim();
         SSOProvider ssoProvider = application.getSsoProviderDetails();
         if (ssoProvider == null) {
             log.warn("No SSO Configurator details given. Manual setup of SSO Provider required.");
         }
 
         //ssoProvider.setIssuerName(app.getId().getApiName());
```
Comment (sajithab) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java:206]: remove commented code

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js [hunk 5]
```diff
@@ -494,51 +494,52 @@ $(function() {
...
-            		contentType: 'application/json',
-            		data:JSON.stringify(sso_config),
-            		success: function(response) {
-                		console.log("Added SSO config successfully");
-            		},
-            		error: function(response) {
-                		showAlert('Error adding service provider.', 'error');
-            		}
-        	});
+	    var sso_config = {};
```
Comment (lakshani) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js]: apply camelCase

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js [hunk 6]
```diff
@@ -494,51 +494,52 @@ $(function() {
...
+	    var app_name = $('#overview_name').val();
+	    var app_version = $('#overview_version').val();
+	    var app_transport = $('#overview_transports').val();
+	    var app_context = $('#overview_context').val();
+	    var app_provider = $('#overview_provider').val();
+	    var app_allowAnonymous=$('#overview_allowAnonymous').val();
+	    var app_acsURL = $('#overview_acsUrl').val();
+
+	    var claims = [];
+	    var index=0;
```
Comment (sajithab) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js:510]: Formating

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js [hunk 7]
```diff
@@ -494,51 +494,52 @@ $(function() {
...
+	    sso_config.app_context = app_context;
+	    sso_config.app_provider = app_provider;
+	    sso_config.app_allowAnonymous=app_allowAnonymous;
+	    sso_config.app_acsURL = app_acsURL;
+
+        $.ajax({
+            url: '/publisher/api/sso/addConfig',
+            type: 'POST',
+            contentType: 'application/json',
+            data:JSON.stringify(sso_config),
```
Comment (sajithab) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js:535]: Formating

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js [hunk 8]
```diff
@@ -494,51 +494,52 @@ $(function() {
...
+            success: function(response) {
+                console.log("Added SSO config successfully");
```
Comment (lahiru823) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js:537]: Need to remove console.log (instead use a server info log)

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java (hunk 2 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java:124]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/f3131fd198c08449ff724321a95ba67682999cc3

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java (hunk 4 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java:206]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/f3131fd198c08449ff724321a95ba67682999cc3

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java (hunk 3 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java:207]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/f3131fd198c08449ff724321a95ba67682999cc3

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js (hunk 6 above)
Comment (lakshani) [reply features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js:510]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/f3131fd198c08449ff724321a95ba67682999cc3

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js (hunk 7 above)
Comment (lakshani) [reply features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js:535]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/f3131fd198c08449ff724321a95ba67682999cc3 and https://github.com/wso2/carbon-appmgt/pull/539/commits/d8ece34320d7a59025051992c5e46fc5fa96518c

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js (hunk 8 above)
Comment (lakshani) [reply features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js:537]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/d8ece34320d7a59025051992c5e46fc5fa96518c.
Removed console log. This is a js file runs on browser, so AFAIK, we cannot user server info log. The functionality of this log has handled by Java level.

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java (hunk 2 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/idp/sso/configurator/IS500SAMLSSOConfigurator.java:124]: Formatting issues are fixed in https://github.com/wso2/carbon-appmgt/pull/539/files

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js (hunk 5 above)
Comment (lakshani) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/js/logic/create.asset.js:497]: Formatting issues are fixed in https://github.com/wso2/carbon-appmgt/pull/539/files

**Pull request merged**